verify-go.sh

```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+
+set -e
+
+gen_checksum()
+{
+    find ./gen -type f -exec md5sum {} \; | sort -k 2 | md5sum
+}
+
+print_checksum_details()
+{
+    find ./gen -type f -exec md5sum {} \; | sort -k 2
+}
+
+print_checksum_details
+CHECKSUM_BEFORE=$(gen_checksum)
+echo "checksum before: ${CHECKSUM_BEFORE}"
+make gen-go
```

Comment:
Now we are running `gen-go` twice in CI - here and in https://github.com/open-telemetry/opentelemetry-proto/blob/f09a72cf8c520c6974cea8cb2f7907e34fd81fe9/makefile#L19

The one inside `ci` can be removed if you think that's worth it.
OTOH, if it doesn't add to much extra time, it might still be useful for `make ci` to generate all files at once, e.g. when testing on local machine.